packages/uicore/src/components/InputWithIdentifier/InputWithIdentifier.tsx

```diff
@@ -57,12 +57,24 @@ export interface InputWithIdentifierProps {
    * identifier changes, either directly or due to changes in name field
    */
   onIdentifierChangeCallback?: (identifier: string) => void
+  /**
+   * Allow hyphen in identifier
+   * @default false
+   */
+  allowHyphen?: boolean
 }
 
 // https://harness.atlassian.net/wiki/spaces/CDNG/pages/736200458/Entity+Identifier
-export function getIdentifierFromName(str: string): string {
+export function getIdentifierFromName(str: string, allowHyphen: boolean): string {
```

Comment:
Please add jest test cases for this method.

Let's also ensure to update this doc: https://harness.atlassian.net/wiki/spaces/CDNG/pages/736200458/Entity+Identifier